examples/cpp_execution/InferenceSetIOBufferExample.cpp

```diff
@@ -0,0 +1,511 @@
+        {
+            return std::make_shared<QBufferWrapper>(it->size);
+        }
+
+        throw std::runtime_error(
+            "Buffer mapping of Input Type not found for buffer named : " +
+            bufferName);
+    }
+
+    [[nodiscard]] shQBufferWrapper
+    createDecodeBuffer(const std::string &bufferName,
+                       const qaic::rt::BufferMappings &allBufferMappings)
+    {
+        auto it =
+            std::find_if(allBufferMappings.begin(), allBufferMappings.end(),
+                         [&bufferName](const qaic::rt::BufferMapping &bufferMapping)
+                         {
+                             return (bufferName == bufferMapping.bufferName);
+                         });
+        if (it != allBufferMappings.end())
+        {
+            return std::make_shared<QBufferWrapper>(1);
+        }
+
+        throw std::runtime_error(
+            "Buffer mapping of Decode Input Type not found for buffer named : " +
+            bufferName);
+    }
+
+    /**
+     * Populate input, output vectors with QBuffer information
+     * @param outputBuffers Vector to use in case this is an output instance
+     * @param nextTokenIds Vector to store output[logits]
+     */
+    std::vector<int64_t> get_logits_from_output_buffers(
+        std::vector<QBuffer> &outputBuffers,
+        std::vector<int64_t> &nextTokenIds)
+    {
+        auto rawOPBufPtr = outputBuffers[outputBuffers.size() - 1].buf;
+        const float *bufferOT = reinterpret_cast<const float *>(rawOPBufPtr);
+        int size_of_logits = outputBuffers[outputBuffers.size() - 1].size / sizeof(float);
+
+        // Calculate the index of the maximum element
+        auto maxElementIter = std::max_element(bufferOT, bufferOT + size_of_logits);
+        int maxElementIndex = std::distance(bufferOT, maxElementIter);
+
+        nextTokenIds.push_back(maxElementIndex);
+        std::vector<int64_t> logits({maxElementIndex});
+        return logits;
+    }
+
+    /**
+     * Given a Input buffer and size, populate it with inputs data(input_ids or position ids)
+     * @param inputBuffer buffer to populate
+     * @param tokenVector vector to fill input Buffer
+     */
+    void populateBuffer(QBuffer &inputBuffer,
+                        const std::vector<int64_t> &tokenVector)
+    {
+        size_t token_size_bytes = tokenVector.size() * sizeof(int64_t);
+        if (inputBuffer.size < token_size_bytes)
+        {
+            delete[] inputBuffer.buf;
+            inputBuffer.buf = new uint8_t[token_size_bytes];
+            inputBuffer.size = token_size_bytes;
+        }
+        std::copy_n(reinterpret_cast<const uint8_t *>(tokenVector.data()),
+                    token_size_bytes, inputBuffer.buf);
+
+        if (inputBuffer.size > token_size_bytes)
+        {
+            std::memset(inputBuffer.buf + token_size_bytes, 0,
+                        inputBuffer.size - token_size_bytes);
+        }
+    }
+
+    template <typename T>
```

Comment:
this function will not be used during inference, so you can keep it in comments or remove later.